core/src/main/scala/org/apache/spark/HeartbeatReceiver.scala

```diff
@@ -62,14 +62,15 @@ private[spark] class HeartbeatReceiver(sc: SparkContext)
 
   // "spark.network.timeout" uses "seconds", while `spark.storage.blockManagerSlaveTimeoutMs` uses
   // "milliseconds"
-  private val executorTimeoutMs = sc.conf.getOption("spark.network.timeout").map(_.toLong * 1000).
-    getOrElse(sc.conf.getLong("spark.storage.blockManagerSlaveTimeoutMs", 120000))
+  private val executorTimeoutMs = Utils.timeStringAsMs(
+    sc.conf.get("spark.storage.blockManagerSlaveTimeoutMs", 
+      s"${Utils.timeStringAsS(sc.conf.get("spark.network.timeout", "120s"))}s"))
 
   // "spark.network.timeoutInterval" uses "seconds", while
   // "spark.storage.blockManagerTimeoutIntervalMs" uses "milliseconds"
-  private val checkTimeoutIntervalMs =
-    sc.conf.getOption("spark.network.timeoutInterval").map(_.toLong * 1000).
-      getOrElse(sc.conf.getLong("spark.storage.blockManagerTimeoutIntervalMs", 60000))
+  private val checkTimeoutIntervalMs = Utils.timeStringAsMs(
+    sc.conf.get("spark.storage.blockManagerTimeoutIntervalMs", 
```

Comment:
Hm but the block manager timeout now has precedence. The network prop did before. Right? I can't see how this isnt different. Or else I'm really blind now.